packages/e2e/e2e-graph/tests/transaction.test.ts

```diff
@@ -462,3 +482,67 @@ test.describe('Query: getTransactions', () => {
     });
   });
 });
+
+test.describe('Subscription: getTransactions', () => {
+  test('Subscriptions: getTransactions - Subscribe to transactions by requestKey', async ({}) => {
+    let account: IAccount;
+
+    await test.step('create a source account on chains 0 and 1 and a target account on chain 1.', async () => {
+      account = await generateAccount(1, ['0', '1']);
+
+      const txTask = transferCreate(
+        {
+          sender: {
+            account: sender00Account.account,
+            publicKeys: sender00Account.keys.map(
+              (keyPair: IKeyPair) => keyPair.publicKey,
+            ),
+          },
+          receiver: {
+            account: account.account,
+            keyset: {
+              keys: account.keys.map((keyPair) => keyPair.publicKey),
+              pred: 'keys-all',
+            },
+          },
+          amount: '100',
+          chainId: account.chains[0],
+        },
+        {
+          host: devnetHost,
+          defaults: {
+            networkId: networkId,
+          },
+          sign: createSignWithKeypair(sender00Account.keys),
+        },
+      );
+      const preflightResponse = await txTask.executeTo('preflight');
+      const query = getTransactionsByRequestKeySubscription(
+        preflightResponse.reqKey,
+        account.chains[0],
+      );
+      const subscription = wsClient.iterate(query);
+      txTask.executeTo();
+      const firstEvent = (await subscription.next()).value.data;
+      expect(firstEvent).toEqual({
+        transaction: {
+          result: {
+            __typename: 'TransactionMempoolInfo',
+            status: 'Pending',
+          },
+        },
+      });
+
+      const secondEvent = (await subscription.next()).value.data;
+      expect(secondEvent).toEqual({
+        transaction: {
+          result: {
+            __typename: 'TransactionResult',
+            badResult: null,
+            goodResult: JSON.stringify('Write succeeded'),
+          },
+        },
+      });
```

Comment:
Are we not playing with race conditions here? So is it guaranteed that the 2nd event will for sure be the transaction result? 
This will only work if the from the `const subscription = wsClient.iterate(query);` statement to the next `txTask.executeTo()` the time difference of when they are executed is less that the time it takes for the subscription to sent out a second event. Correct me if I am wrong @Ghislain89 , just doing a check 😄

Reply:
Theoretically, I think you're right. Practically speaking I believe the risk to be negligible in this case. The way I understand the subscription, is that an event is only emitted when a response (== change in data) is returned.  

By not awaiting the `txTask.executeTo()` the test just continues on, the first event that is emitted should _always_ be a Pending result, as the subscription was already active before we actually sent the tx to the chain. The second change in data should then always be a finished transaction given that we're using the instant mining sandbox container. 

If I understood correctly how a subscription works, if there is no change in response compared to the first event, no new event is emitted: meaning the second response should always be the succes message.

Reply:
Hmm... I see, and I think you might be right and what we do currently is actually not correct. At the moment we return undefined on every iteration. I could change it so it only returns something if its different from the previous event